File: src/main/java/me/diax/diax/commands/information/Ping.java
package me.diax.diax.commands.information;

import me.diax.comportment.jdacommand.Command;
import me.diax.comportment.jdacommand.CommandAttribute;
import me.diax.comportment.jdacommand.CommandDescription;
import me.diax.diax.util.Emote;
import net.dv8tion.jda.core.entities.Message;

import java.util.Random;

@CommandDescription(
        name = "ping",
        triggers = {"ping", "pang", "peng", "pong", "pung"},
        description = "Shows the message response time and api ping.",
        attributes = @CommandAttribute(key = "private")
)
public class Ping implements Command {

    @Override
    public void execute(Message message, String s) {
        long start = System.currentTimeMillis();
        message.getChannel().sendTyping().queue(typing -> {
            long ping = System.currentTimeMillis() - start;
            message.getChannel().sendMessage(Emote.PING_PONG + " - ***P" + new String[]{"a", "e", "i", "o", "u"}[new Random().nextInt(5)] + "ng!***\n" + this.pingToEmote(ping) + " Response: " + ping + "ms\n" + Emote.BOOKS + " API: " + message.getJDA().getPing() + "ms").queue();
        });
    }

    public String pingToEmote(long ping) {
        if (ping == 69) return Emote.EGGPLANT;
        if (ping <= 0) return Emote.UPSIDE_DOWN;
        if (ping <= 10) return Emote.SMILE;
        if (ping <= 100) return Emote.SMILEY;
        if (ping <= 200) return Emote.SLIGHT_SMILE;
        if (ping <= 300) return Emote.NEUTRAL_FACE;
        if (ping <= 400) return Emote.CONFUSED;
        if (ping <= 500) return Emote.SLIGHT_FROWN;
        if (ping <= 600) return Emote.FROWNING2;
        if (ping <= 700) return Emote.WORRIED;
        if (ping <= 800) return Emote.DISAPPOINTED;
        if (ping <= 900) return Emote.SOB;
        if (ping <= 1600) return Emote.FIRE;
        if (ping <= 10000) return Emote.FIRE + Emote.FIRE;
        return Emote.FIRE + Emote.FIRE + Emote.FIRE;
    }
}

File: src/main/kotlin/me/diax/diax/commands/information/Ping.kt
package me.diax.diax.commands.information

import me.diax.comportment.jdacommand.Command
import me.diax.comportment.jdacommand.CommandDescription
import me.diax.diax.util.Emote.BOOKS
import me.diax.diax.util.Emote.CONFUSED
import me.diax.diax.util.Emote.DISAPPOINTED
import me.diax.diax.util.Emote.EGGPLANT
import me.diax.diax.util.Emote.FIRE
import me.diax.diax.util.Emote.FROWNING2
import me.diax.diax.util.Emote.NEUTRAL_FACE
import me.diax.diax.util.Emote.PING_PONG
import me.diax.diax.util.Emote.SLIGHT_FROWN
import me.diax.diax.util.Emote.SLIGHT_SMILE
import me.diax.diax.util.Emote.SMILE
import me.diax.diax.util.Emote.SMILEY
import me.diax.diax.util.Emote.SOB
import me.diax.diax.util.Emote.UPSIDE_DOWN
import me.diax.diax.util.Emote.WORRIED
import me.diax.diax.util.style.Embed
import net.dv8tion.jda.core.entities.Message
import java.util.*

@CommandDescription(
    name = "ping",
    triggers = ["ping", "pang", "peng", "pong", "pung"]
)
class Ping : Command {

    override fun execute(message: Message, s: String) {
        val start = System.currentTimeMillis()
        message.channel.sendTyping().queue { _ ->
            val ping = System.currentTimeMillis() - start
            message.channel.sendMessage(Embed.themed().setDescription("$PING_PONG - ***P${arrayOf("a", "e", "i", "o", "u")[Random().nextInt(5)]}ng!***\n${this.pingToEmote(ping)} Response: ${ping}ms\n$BOOKS API: ${message.jda.ping}ms").build()).queue()
        }
    }

    private fun pingToEmote(ping: Long): String {
        return when {
            ping == 69L -> EGGPLANT
            ping <= 0 -> UPSIDE_DOWN
            ping <= 10 -> SMILE
            ping <= 100 -> SMILEY
            ping <= 200 -> SLIGHT_SMILE
            ping <= 300 -> NEUTRAL_FACE
            ping <= 400 -> CONFUSED
            ping <= 500 -> SLIGHT_FROWN
            ping <= 600 -> FROWNING2
            ping <= 700 -> WORRIED
            ping <= 800 -> DISAPPOINTED
            ping <= 900 -> SOB
            ping <= 1600 -> FIRE
            ping <= 10000 -> FIRE * 2
            else -> FIRE * 3
        }
    }

    private operator fun String.times(i: Int): String {
        val b = StringBuilder()
        for (any in 0..i) b.append(this)
        return b.toString()
    }
}
